prow/plank/controller.go

```diff
@@ -135,6 +135,25 @@ func (c *Controller) canExecuteConcurrently(pj *prowapi.ProwJob) bool {
 		c.log.WithFields(pjutil.ProwJobFields(pj)).Debugf("Not starting another instance of %s, already %d running.", pj.Spec.Job, numPending)
 		return false
 	}
+
+	var olderMatchingPJs []prowapi.ProwJob
+	for _, foundPJ := range c.pjs {
+		if foundPJ.Status.State != prowapi.TriggeredState {
+			continue
+		}
+		if foundPJ.Spec.Job != pj.Spec.Job {
+			continue
+		}
+		if foundPJ.CreationTimestamp.Before(&pj.CreationTimestamp) {
+			olderMatchingPJs = append(olderMatchingPJs, foundPJ)
+		}
+	}
+	if numPending+len(olderMatchingPJs) >= pj.Spec.MaxConcurrency {
+		c.log.WithFields(pjutil.ProwJobFields(pj)).Debugf("Not starting another instance of %s, already %d running and %d older instances waiting",
```

Comment:
nit: This message was confusing to see in the logs because the individual numbers did not correspond with the `max_concurrency` field value. The message should be more clear now that the individual numbers correctly sum to the value of `max_concurrency`, but it would probably still be helpful to include the value of `max_concurrency` in this message for clarity.